agent/06_agent_create_cluster.sh

```diff
@@ -23,9 +23,55 @@ function create_pxe_files() {
 function create_image() {
     local asset_dir=${1}
     local openshift_install=${2}
+
+    if [ "${AGENT_USE_APPLIANCE_MODEL}" == true ]; then
+      create_factory_image
+    else
+      create_automated_image
+    fi
+}
+
+function create_automated_image() {
     "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create image
 }
 
+function create_factory_image() {
+    config_image_drive="sdd"
+
+    # The command to create the config-image must be run out of a separate asset directory using same assets
+    mkdir -p ${config_image_dir}
+    cp ${asset_dir}/*.yaml ${config_image_dir}
+
+    # Create the unconfigured ingition and include it in an ISO
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create cluster-manifests
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create unconfigured-ignition
+    base_iso_url=$(oc adm release info --registry-config "$PULL_SECRET_FILE" --image-for=machine-os-images --insecure=true $OPENSHIFT_RELEASE_IMAGE)
+    oc image extract --path /coreos/coreos-x86_64.iso:$HOME/.cache/agent/image_cache --registry-config "$PULL_SECRET_FILE" --confirm $base_iso_url
+    local agent_iso_abs_path="$(realpath "${OCP_DIR}")"
+    podman run --privileged --rm -v /run/udev:/run/udev -v "${agent_iso_abs_path}:${agent_iso_abs_path}" -v "$HOME/.cache/agent/image_cache/:$HOME/.cache/agent/image_cache/" quay.io/coreos/coreos-installer:release iso ignition embed -f -i "${agent_iso_abs_path}/unconfigured-agent.ign" -o "${agent_iso_abs_path}/agent.iso" $HOME/.cache/agent/image_cache/coreos-x86_64.iso
+
+    if [ "${AGENT_APPLIANCE_HOTPLUG}" != true ]; then
+        create_config_image
+    fi
+}
+
+function create_config_image() {
+
+    "${openshift_install}" --log-level=debug --dir="${config_image_dir}" agent create config-image
+
+    # Copy the auth files to OCP_DIR so wait-for command can access it
+    cp -r ${config_image_dir}/auth ${asset_dir}
```

Comment:
Isn't it cleaner to just change the dir in wait_for_cluster_ready func.
I.e. in config-image flow use --dir="${config_image_dir}" instead of --dir="${OCP_DIR}"

Reply:
Good point. Changed